integration_test/generate_expected_metrics.go

```diff
@@ -65,41 +60,33 @@ func run() error {
 	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
 	defer cancel()
 
-	listMetrics, err := listAllMetricsWithLabels(ctx, project)
+	allMetrics, err := listAllMetricsByApp(ctx, project)
 	if err != nil {
 		return err
 	}
 
-	metricsByApp := expectedMetricsByApp(listMetrics)
-	for app, newMetrics := range metricsByApp {
+	for app, newMetrics := range allMetrics {
 		log.Printf("Processing %d metrics for %s...\n", len(newMetrics), app)
-		metricsToWrite := make([]expectedMetric, 0)
 		existingMetrics, readErr := readExpectedMetrics(app)
 		if readErr != nil {
 			err = multierr.Append(err, readErr)
 			continue
 		}
-		// Write existing metrics first, updating them if needed
-		for _, existingMetric := range existingMetrics {
-			if newMetric := findMetric(newMetrics, existingMetric.Type); newMetric != nil {
-				existingMetric = mergeMetric(existingMetric, *newMetric)
-			}
-			metricsToWrite = append(metricsToWrite, existingMetric)
-		}
-		// Write exclusively new metrics last
+		// For each new metric, either update the corresponding existing metric,
+		// or add it.
 		for _, newMetric := range newMetrics {
-			if findMetric(existingMetrics, newMetric.Type) == nil {
-				metricsToWrite = append(metricsToWrite, newMetric)
+			if existingMetric, ok := existingMetrics[newMetric.Type]; ok {
```

Comment:
You cannot take the address of a map value :(

Reply:
Err, I should have known that. However, you could still apply my suggestion with a functional update, e.g.:
```
			existingMetrics[newMetric.Type] = updateMetric(existingMetrics[newMetric.Type], newMetric)
```
(you would need to add a test in `updateMetric` for `toUpdate.Type == ""` and just assign `withValuesFrom.Type`). This would avoid a whole lot of pointers…

But that's still optional, so up to you.

Reply:
I do like that suggestion. Since a follow-up PR is already coming I'll defer it to that one.